scripts/prepare_sector_network.py

```diff
@@ -488,22 +488,9 @@ def add_links_elec_routing_new_H2_pipelines():
         # Order buses to detect equal pairs for bidirectional pipelines
         buses_ordered = h2_links.apply(lambda p: sorted([p.bus0, p.bus1]), axis=1)
 
-        if snakemake.config["build_osm_network"]["force_ac"]:
-            # Appending string for carrier specification '_AC'
-            h2_links["bus0"] = buses_ordered.str[0] + "_AC"
-            h2_links["bus1"] = buses_ordered.str[1] + "_AC"
-
-            # # Conversion of GADM id to from 3 to 2-digit
-            # h2_links["bus0"] = (
-            #     h2_links["bus0"]
-            #     .str.split(".")
-            #     .apply(lambda id: three_2_two_digits_country(id[0]) + "." + id[1])
-            # )
-            # h2_links["bus1"] = (
-            #     h2_links["bus1"]
-            #     .str.split(".")
-            #     .apply(lambda id: three_2_two_digits_country(id[0]) + "." + id[1])
-            # )
+        # Appending string for carrier specification '_AC', because hydrogen has _AC in bus names
+        h2_links["bus0"] = buses_ordered.str[0] + "_AC"
+        h2_links["bus1"] = buses_ordered.str[1] + "_AC"

```

Comment:
Can you explain this proposal? Have you tested the feature with and without force_ac in case?

Reply:
Hi, @davide-f. Thanks for the comment. Currently, hydrogen buses are added for `spatial.node`:
https://github.com/pypsa-meets-earth/pypsa-earth/blob/e386102552dc4471115432b5980e08125238b5f7/scripts/prepare_sector_network.py#L193-L200
It comes from `pop_layout.index`, which have buses names taken from `regions_onshore_elec_s_10.geojson`. So buses have names in form of `US.1_1_AC`. In other words, all buses are in form of `US.10_1_AC {technology}` (hydrogen buses are `US.1_1_AC H2`). The problem is that by default `bus0` and `bus1` of `h2_links` (and `buses_ordered`) are in a form of `US.1_1` without any `_AC` part. As a result, there were errors of having links with no defined buses (since those links are not attached to H2 buses with _AC parts in names). 

I have tested it with `force_ac: false`. I can test for `force_ac: true` too.

Reply:
The warning will be something like this. These links are connected to non-existing buses.
![image](https://github.com/user-attachments/assets/0215ec67-63df-4fa3-a113-6ee2c425aebd)

Reply:
For NG, both force_ac false and true worked fine. But I did not compared the networks.

Reply:
Great @yerbol-akhmetov :D Thanks!